src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationExecutor.kt

```diff
@@ -56,9 +56,14 @@ class ShardReplicationExecutor(executor: String, private val clusterService : Cl
     }
 
     override fun getAssignment(params: ShardReplicationParams, clusterState: ClusterState) : Assignment {
-        val primaryShard = clusterState.routingTable().shardRoutingTable(params.followerShardId).primaryShard()
-        if (!primaryShard.active()) return SHARD_NOT_ACTIVE
-        return Assignment(primaryShard.currentNodeId(), "node with primary shard")
+        try {
+            val primaryShard = clusterState.routingTable().shardRoutingTable(params.followerShardId).primaryShard()
```

Comment:
For testing, will _stop API clear these entries when the shards are not active?

Reply:
For this we are taking another issue
https://github.com/opensearch-project/cross-cluster-replication/issues/903